libsel4/arch_include/x86/sel4/arch/vmenter.h

```diff
@@ -6,36 +6,48 @@
 
 #pragma once
 
-/*
- * When performing a seL4_SysVMEnter seL4 expects certain guest state to be
- * placed in the message registers. These defines indicates which MRs hold
- * which values. Whenever a VMEnter returns seL4 will also fill these registers
- * with the corresponding guest state
- */
-#define SEL4_VMENTER_CALL_EIP_MR 0
-#define SEL4_VMENTER_CALL_CONTROL_PPC_MR 1
-#define SEL4_VMENTER_CALL_CONTROL_ENTRY_MR 2
-
-/*
- * In addition to the above message registers, if a VMEnter results in
- * a fault the following constants describe the contents of the message
- * registers that contain fault specific information
- */
-#define SEL4_VMENTER_FAULT_REASON_MR 3
-#define SEL4_VMENTER_FAULT_QUALIFICATION_MR 4
-#define SEL4_VMENTER_FAULT_INSTRUCTION_LEN_MR 5
-#define SEL4_VMENTER_FAULT_GUEST_PHYSICAL_MR 6
-#define SEL4_VMENTER_FAULT_RFLAGS_MR 7
-#define SEL4_VMENTER_FAULT_GUEST_INT_MR 8
-#define SEL4_VMENTER_FAULT_CR3_MR 9
-#define SEL4_VMENTER_FAULT_EAX 10
-#define SEL4_VMENTER_FAULT_EBX 11
-#define SEL4_VMENTER_FAULT_ECX 12
-#define SEL4_VMENTER_FAULT_EDX 13
-#define SEL4_VMENTER_FAULT_ESI 14
-#define SEL4_VMENTER_FAULT_EDI 15
-#define SEL4_VMENTER_FAULT_EBP 16
+typedef enum kernel_fault_context {
+    /*
+     * When performing a seL4_SysVMEnter seL4 expects certain guest state to be
+     * placed in the message registers. These defines indicates which MRs hold
+     * which values. Whenever a VMEnter returns seL4 will also fill these registers
+     * with the corresponding guest state
+     */
+    SEL4_VMENTER_CALL_EIP_MR,
+    SEL4_VMENTER_CALL_CONTROL_PPC_MR,
+    SEL4_VMENTER_CALL_CONTROL_ENTRY_MR,
 
+    /*
+     * In addition to the above message registers, if a VMEnter results in
+     * a fault the following constants describe the contents of the message
+     * registers that contain fault specific information
+     */
+    SEL4_VMENTER_FAULT_REASON_MR,
+    SEL4_VMENTER_FAULT_QUALIFICATION_MR,
+    SEL4_VMENTER_FAULT_INSTRUCTION_LEN_MR,
+    SEL4_VMENTER_FAULT_GUEST_PHYSICAL_MR,
+    SEL4_VMENTER_FAULT_RFLAGS_MR,
+    SEL4_VMENTER_FAULT_GUEST_INT_MR,
+    SEL4_VMENTER_FAULT_CR3_MR,
+    SEL4_VMENTER_FAULT_EAX,
+    SEL4_VMENTER_FAULT_EBX,
+    SEL4_VMENTER_FAULT_ECX,
+    SEL4_VMENTER_FAULT_EDX,
+    SEL4_VMENTER_FAULT_ESI,
+    SEL4_VMENTER_FAULT_EDI,
+    SEL4_VMENTER_FAULT_EBP,
+#ifdef CONFIG_ARCH_X86_64
+    SEL4_VMENTER_FAULT_R8,
+    SEL4_VMENTER_FAULT_R9,
+    SEL4_VMENTER_FAULT_R10,
+    SEL4_VMENTER_FAULT_R11,
+    SEL4_VMENTER_FAULT_R12,
+    SEL4_VMENTER_FAULT_R13,
+    SEL4_VMENTER_FAULT_R14,
+    SEL4_VMENTER_FAULT_R15,
+#endif
+    SEL4_VMENTER_NUM_FAULT_MSGS,
+} kernel_fault_context_t;
```

Comment:
This enum should go into a CONFIG_VTX ifdef to hide it from the preprocessor.  This should resolve the preprocess failures.

Reply:
Yep.